specs/electra/beacon-chain.md

```diff
@@ -631,12 +634,19 @@ def switch_to_compounding_validator(state: BeaconState, index: ValidatorIndex) -
 
 ```python
 def queue_excess_active_balance(state: BeaconState, index: ValidatorIndex) -> None:
+    validator = state.validators[index]
     balance = state.balances[index]
     if balance > MIN_ACTIVATION_BALANCE:
         excess_balance = balance - MIN_ACTIVATION_BALANCE
         state.balances[index] = MIN_ACTIVATION_BALANCE
-        state.pending_balance_deposits.append(
-            PendingBalanceDeposit(index=index, amount=excess_balance)
+        state.pending_deposits.append(
+            PendingDeposit(
+                epoch=get_current_epoch(state),
+                pubkey=validator.pubkey,
+                withdrawal_credentials=validator.withdrawal_credentials,
```

Comment:
Are `withdrawal_credentials` necessary to be saved here? 

- If yes, there could be a race where a `BLSToExecutionChange` gets processed while the `PendingDeposit` is in the queue, changing the withdrawal credentials of the validator while the `PendingDeposit` retains the stale entry.
- If no, we could just set this to a zero value (or to `None` with EIP-7688 optionals).

Reply:
It isn’t necessary so I think that setting it to zero value should be fine